Revit_Core_Engine/Query/Perimeter.cs

```diff
@@ -44,24 +47,14 @@ public static List<PolyCurve> Perimeter(this SpatialElement spatialElement, Revi
             if (spatialElement == null)
                 return null;
 
-            IList<IList<BoundarySegment>> boundarySegments = spatialElement.GetBoundarySegments(new SpatialElementBoundaryOptions());
-            if (boundarySegments == null)
-                return null;
+            Autodesk.Revit.DB.Face face = spatialElement.Faces(new Options()).Where(x => Math.Abs((x as PlanarFace).FaceNormal.Z + 1) < Tolerance.MacroDistance).FirstOrDefault();
```

Comment:
Why are you changing the methodology of extracting the perimeter here? Is there a bug in the current algorithm? I am asking because we need to be very careful with this method as is it used by multiple convert methods, which means the change will require significant testing and probably agreement from S&P.

Reply:
I restored this method to previous version.